util-core/src/main/scala/com/twitter/util/Batcher.scala

```diff
@@ -0,0 +1,30 @@
+package com.twitter.util
+
+import java.util.logging.{Level, Logger}
+
+import scala.collection.mutable
+
+/** Provides a clean, lightweight interface for controlling a BatchExecutor
```

Comment:
As you mentioned before, `BatchExecutor` is an internal implementation detail, so when giving folks instructions in how to use `Batcher`, we shouldn't ask them to understand `BatchExecutor`.